Library/Homebrew/download_strategy.rb

```diff
@@ -500,19 +495,20 @@ def resolve_url_basename_time_file_size(url, timeout: nil)
       File.basename(filename)
     end
 
-    filenames = lines.map(&parse_content_disposition).compact
+    filenames = parsed_headers.flat_map do |headers|
+      next [] if (header = headers["content-disposition"]).nil?
+
+      [*parse_content_disposition.call("Content-Disposition: #{header}")]
+    end
 
-    time =
-      lines.map { |line| line[/^Last-Modified:\s*(.+)/i, 1] }
-           .compact
+    time = parsed_headers
+           .flat_map { |headers| [*headers["last-modified"]] }
            .map { |t| t.match?(/^\d+$/) ? Time.at(t.to_i) : Time.parse(t) }
            .last
 
-    file_size =
-      lines.map { |line| line[/^Content-Length:\s*(\d+)/i, 1] }
-           .compact
-           .map(&:to_i)
-           .last
+    file_size = parsed_headers
+                .flat_map { |headers| [*headers["content-length"]&.to_i] }
```

Comment:
```suggestion
                .flat_map { |headers| [*headers["content-length"]&.to_i] }
                .compact
```

is this needed if `flat_map` may be `nil` sometimes?

Reply:
`flat_map` is never `nil`, only empty, so we don't need `compact`.